crates/testing/tests/da_task.rs

```diff
@@ -80,7 +87,20 @@ async fn test_da_task() {
             output.insert(SequencingHotShotEvent::DAVoteSend(vote), 1);
         }
     }
+    output.insert(
+        SequencingHotShotEvent::VidDisperseSend(
+            VidDisperse {
+                view_number: message.data.view_number,
+                shares,
+                common,
+            },
+            pub_key,
+        ),
+        1,
+    );
+    // TODO why does this test pass without a `VidVoteSend` event here? https://github.com/EspressoSystems/HotShot/issues/1697
     output.insert(SequencingHotShotEvent::DAProposalRecv(message, pub_key), 1);
+    // TODO why does this test pass without a `VidDisperseRecv` event here? https://github.com/EspressoSystems/HotShot/issues/1697
```

Comment:
In that case, why does this test need a `DAProposalRecv` event in the output but not a `VidDisperseRecv`? I expect either both are needed or neither.

Reply:
Though neither of them is sent by the DA task, `DAProposalRecv` is in the input, so it will also be in the output. If we don't have `DAProposalRecv` in the input, then indeed, it won't be expected in the output.

Reply:
I see. Indeed, the test passes if I add `VidDisperseRecv` to both `input` and `output`. But what's the point? Do these additions hit any new code paths? Would the test still function correctly if I were to delete both `VidDisperseRecv` and `DAProposalRecv` from both `input` and `output`?

Reply:
I think the answer is that `VidVoteSend` does not appear unless `VidDisperseRecv` is in `input` and `output`. In other words, if `VidDisperseRecv` is missing then we do not test VID votes. I suspect the same is true for `DAProposalRecv` and `DAVoteSend`. See https://github.com/EspressoSystems/HotShot/pull/1684#discussion_r1323512874